src/document/HolonomicPathStore.ts

```diff
@@ -807,17 +798,19 @@ export const HolonomicPathStore = types
       // when mobx first runs it to determine dependencies.
       staleDisposer = reaction(
         () => {
-          toJS(self.waypoints);
-          toJS(self.constraints);
-          // does not need toJS to do a deep check on this, since it's just a boolean
-          self.usesControlIntervalGuessing;
-          toJS(self.obstacles);
-          return true;
+          // Reaction needs the return value to change,
+          // so we can't just access the values and do nothing with them
+          
+          return {
+            waypoints: toJS(self.waypoints),
+            constraints: toJS(self.constraints),
+            // does not need toJS to do a deep check on this, since it's just a boolean
+            guessing: self.usesControlIntervalGuessing,
+            obstacles: toJS(self.obstacles)
```

Comment:
should this be a type?

Reply:
No, it's not needed since this structure only exists in one spot.